src/sentry/dynamic_sampling/tasks.py

```diff
@@ -320,3 +327,96 @@ def process_transaction_biases(project_transactions: ProjectTransactions) -> Non
     schedule_invalidate_project_config(
         project_id=project_id, trigger="dynamic_sampling_prioritise_transaction_bias"
     )
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.tasks.sliding_window_rebalancing",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=2 * 60 * 60,  # 2 hours
+    time_limit=2 * 60 * 60 + 5,
+)  # type: ignore
+def sliding_window_rebalancing() -> None:
+    metrics.incr("sentry.dynamic_sampling.tasks.sliding_window_rebalancing.start", sample_rate=1.0)
+    with metrics.timer("sentry.dynamic_sampling.tasks.sliding_window_rebalancing", sample_rate=1.0):
+        for orgs in get_orgs_with_project_counts_without_modulo(
+            MAX_ORGS_PER_QUERY, MAX_PROJECTS_PER_QUERY
+        ):
+            for (
+                org_id,
+                projects_with_total_root_count,
+            ) in fetch_projects_with_total_root_transactions_count(org_ids=orgs).items():
+                apply_sliding_window_rebalancing_on_projects.delay(
+                    org_id, projects_with_total_root_count
+                )
+
+
+@instrumented_task(
+    name="sentry.dynamic_sampling.apply_sliding_window_rebalancing_on_projects",
+    queue="dynamicsampling",
+    default_retry_delay=5,
+    max_retries=5,
+    soft_time_limit=25 * 60,  # 25 mins
+    time_limit=2 * 60 + 5,
+)  # type: ignore
+def apply_sliding_window_rebalancing_on_projects(
+    org_id: OrganizationId,
+    projects_with_total_root_count: Sequence[Tuple[ProjectId, int]],
+) -> None:
+    with metrics.timer(
+        "sentry.tasks.dynamic_sampling.apply_sliding_window_rebalancing_on_projects.core"
+    ):
+        adjust_base_sample_rate_per_project(org_id, projects_with_total_root_count)
+
+
+def adjust_base_sample_rate_per_project(
+    org_id: int,
+    projects_with_total_root_count: Sequence[Tuple[ProjectId, int]],
+) -> None:
+    """
+    Adjusts the base sample rate per project by considering its volume and how it fits w.r.t. to the sampling tiers
+    defined on the billing side.
+    """
+    projects_with_rebalanced_sample_rate = []
+    for project_id, total_root_count in projects_with_total_root_count:
+        forecasted_volume = get_forecasted_monthly_volume(daily_volume=total_root_count)
+        if forecasted_volume is None:
```

Comment:
If we have no volume, should we instead bump the sample rate to 100%? That would help with onboarding, where we usually want to see the first event through.

If `None` here means something else, like we were unable to retrieve stats etc, then we shouldn't do this, of course.

Reply:
This function will return `None` in case of errors. The value will be always > 0, assuming a positive root count and a value > 0 which for our query is.